src/robot/running/arguments/typeconverter.py

```diff
@@ -39,12 +40,17 @@ def convert(self, positional, named):
     def _convert(self, name, value):
         if name not in self._argspec.types or not is_string(value):
             return value
+        if value.upper() == 'NONE':
+            return None
         type_ = self._argspec.types[name]
+        if isinstance(type_, EnumMeta):
+            try:
+                return type_[value]
+            except KeyError:
+                self._raise_convert_failed(name, type_.__name__, value)
```

Comment:
Although enums need to be recognized differently than other types, it would be better to have a separate `_conver_enum` helper for conversion. Code here could hen be change to something like:

```python
if isinstance(type_, EnumMeta):
    converter = self._convert_enum
else:
    converter = self._converters.get(type_)
```